### ISSUE TITLE
inference speed

### ISSUE THREAD

**Issue by kitlomer** [fatchord/WaveRNN#67]
Thank you for sharing your code and pretrained models! I tried quick_start and the sound quality is pretty good. I am using the default hparams. However, the inference time is about 90s for generating 2s of sound on my RTX 2080 Ti, which is much slower than reported in their paper. The voc_gen_batched is already set as True in the hparams. Is it normal or should I adjust any other hparams to get a faster speed? 
Thanks a lot.

**Comment by fatchord** [comment]
@kitlomer That sounds totally off. It should be generating at realtime speed on a badass GPU like the 2080ti (jealous by the way!). What version of Pytorch are you using?

**Comment by kitlomer** [comment]
Thanks for your quick reply! It's version 1.0.1. I must have done sth wrong but not sure what... /_\

**Comment by fatchord** [comment]
@kitlomer I've no idea why that's happening. The only thing I can think of is that you might not have the right CUDA version (I'm using CUDA 10.1 for what it's worth).

**Comment by williamluke4** [comment]
I am also having the same issue using a `RTX 2080Ti`. I have check to see if Torch is picking up the `2080Ti` with the code below, which it seems to be doing. I'm currently running on `CUDA 10.1` and `pytorch 1.1.0` 
```
>> import torch
>> torch.cuda.current_device()
0
>> torch.cuda.device(0)
<torch.cuda.device object at 0x0000014C02E44C50>
>> torch.cuda.get_device_name(0)
'GeForce RTX 2080 Ti'
>>torch.cuda.is_available()
True
```

**Comment by williamluke4** [comment]
Don't know if this would help;  https://vimeo.com/334931800

**Comment by acrosson** [comment]
@fatchord I haven't been able to get real-time speed using the `quick_start.py` and i'm running it on a Tesla V100. It seems like it takes at least 14 seconds to generate (batched), regardless of the length of the sentence.

I have Ubuntu 16.04, Python 3.6.8, Cuda 10.1 (+latest CudNN), PyTorch 1.1.0 installed. Does this match your build?

![Screen Shot 2019-05-08 at 5 56 52 PM](https://user-images.githubusercontent.com/4795661/57420181-44497a80-71bb-11e9-842b-db89108880d9.png)

**Comment by williamluke4** [comment]
I have found that changing the following params in `hparams.py` will increase the inference speed albeit with a hit to quality. 
```py
voc_target = 5000               # target number of samples to be generated in each batch entry 11_000
voc_overlap = 100    
```

**Comment by fatchord** [comment]
@williamluke4 @acrosson I can't really do much debugging without an RTX card so all I can do is list what I am running: 

GTX 1080 card
Ubuntu 18
python 3.7.3
torch.__version__ = 1.0.1.post2
CUDA 10.1
CUDNN installed with pytorch
torch.backends.cudnn.version() = 7402 (I'm using the cudnn packaged with pytorch conda installation)

**Comment by kitlomer** [comment]
Thanks @fatchord  I switched from CUDA9 to CUDA10 and it is generating 2s of speech in 6.8s, much better now. Will try tinkering with voc params suggested by you @williamluke4 . Thanks for all the replies!

**Comment by acrosson** [comment]
@fatchord I updated my config to match yours. I got the same results as above. I only saw a speedup when modifying the variables like @williamluke4 mentioned, but still not realtime.

What do you mean when you say real-time? Do you mean the `Gen rate` exceeds 20.5k? or the total time it takes to generate a 1 second audio clip is less than 1 second? 

Would you mind sharing a screenshot of the results you get form `quick_start.py`?

**Comment by CorentinJ** [comment]
How did you get cuda 10.1 with PyTorch? Currently it is shipped with 10.0 and even with conda the latest cudatoolkit packages I see are on 10.0.
